lib/Serialization/Deserialization.cpp

```diff
@@ -4200,7 +4200,8 @@ Expected<Type> ModuleFile::getTypeChecked(TypeID TID) {
     SubstitutionMap subMap = getSubstitutionMap(substitutionsID);
 
     // Look through compatibility aliases that are now unavailable.
```

Comment:
No, this one's still correct as is! Though we don't have any such aliases today.

Reply:
Are you sure? We return the underlying type of the alias (which will have open generic parameters) and not the deserialized underlying type (which already has substitutions applied).

Reply:
Uh…okay, it's not correct as is, but it would be easy to make it do the right thing instead of just bailing out.